src/centralserver/admin-ui/tests/e2e/specs/cs-login.js

```diff
@@ -23,57 +23,53 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-const login = (browser) => {
-  const frontPage = browser.page.loginpage();
-
-  frontPage
-    .clearUsername()
-    .clearPassword()
-    .enterUsername(browser.globals.login_usr)
-    .enterPassword(browser.globals.login_pwd)
-    .signin();
-};
-
 module.exports = {
   tags: ['cs', 'login'],
-  'Security server failed login': (browser) => {
+  before(browser) {
     const frontPage = browser.page.loginpage();
-
-    // Open SUT and check that page is loaded
     frontPage.navigate();
-    browser.waitForElementVisible('//*[@id="app"]');
-
-    // Enter invalid credentials
-    frontPage
-      .enterUsername(browser.globals.login_wrong_usr)
-      .enterPassword(browser.globals.login_wrong_pwd)
-      .signin();
-
-    // Verify there's an error message
-    browser.waitForElementVisible(
-      '//div[contains(@class, "v-messages__message")]',
-    );
-
-    browser.end();
   },
-  'Security server passed login': (browser) => {
-    const frontPage = browser.page.loginpage();
 
-    // Open SUT and check that page is loaded
+  afterEach(browser) {
+    browser.refresh();
+    const frontPage = browser.page.loginpage();
     frontPage.navigate();
-    browser.waitForElementVisible('//*[@id="app"]');
-
-    // Enter valid credentials
-    login(browser);
-
-    // Verify successful login
-    browser.waitForElementVisible('//div[contains(@class, "server-name")]');
-
-    // Test refresh
-    browser
-      .refresh()
-      .waitForElementVisible('//div[contains(@class, "server-name")]');
-
+  },
+  after(browser) {
     browser.end();
   },
+  'Wrong username is rejected': (browser) => {
+    const login = browser.page.loginpage();
+    login
+      .enterUsername('invalid')
+      .enterPassword(browser.globals.login_pwd)
+      .signin()
+      .loginErrorMessageIsShown();
+  },
+  'Wrong password is rejected': (browser) => {
+    const login = browser.page.loginpage();
+    login
+      .enterUsername(browser.globals.login_usr)
+      .enterPassword('invalid')
+      .signin()
+      .loginErrorMessageIsShown();
+  },
+  'Admin login succeeds': (browser) => {
+    const login = browser.page.loginpage();
```

Comment:
there is a way to reduce this repetion by putting `let login;` above `module.exports` and then you need to populate it only once in `before` -block and it will be available for all other blocks afterwards. Here it would eliminate 8 rows from tests and add two to test setup.

Reply:
Rebased the e2e changes in a new pull request: https://github.com/nordic-institute/X-Road/pull/1003/files#diff-ce670a42e96cbab4aaa8d6758f969bec52d4edaf824720e98adb355e30d1ec74